### PR TITLE
Addresses BACKLOG-37841: Change schedule ownership

### PR DESCRIPTION
**Pull request by afrjorge**
Add the ability for an administrator to change the current schedule owner.

@pentaho/millenniumfalcon please review

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 52m 30s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions
```
---
:ok_hand: All tests passed!
 
Tests run: **1694**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5340/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: extensions/src/test/java/org/pentaho/platform/web/http/api/resources/services/SchedulerServiceTest.java [hunk 1]
```diff
@@ -1025,6 +1036,288 @@ public void testUpdateBlockoutException() throws Exception {
     verify( schedulerService, times( 2 ) ).addBlockout( jobScheduleRequest );
   }
 
+  @Test
+  public void testHasLogicalRole() throws Exception {
```
Comment (davidmsantos90) [inline extensions/src/test/java/org/pentaho/platform/web/http/api/resources/services/SchedulerServiceTest.java:1040]: Like the comments inside this unit test (and others below) indicate, shouldn't this be split into different unit tests?
`testHasLogicalRole_nullUser`, `testHasLogicalRole_nullRole`,...

Comment (afrjorge) [reply extensions/src/test/java/org/pentaho/platform/web/http/api/resources/services/SchedulerServiceTest.java:1040]: I agree that it should, I was just following the current file structure. By splitting it makes it easier to identify which test failed. I will rearrange the tests I wrote.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 46m 50s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl extensions
```
---
:ok_hand: All tests passed!
 
Tests run: **1719**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform/job/PR-5340/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by davidmsantos90 — APPROVED** [review]

**Review by mbrasil — APPROVED** [review]
Well done! As always, a nice and clean code. :)

**Comment by buildguy** [comment]
### :x: Build failed in 1m 3s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
assemblies/pentaho-server,extensions,repository,scheduler,user-console
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by afrjorge** [comment]
Scheduler improvements will be merged into the [scheduler-plugin](https://github.com/pentaho/pentaho-platform/tree/scheduler-plugin) branch

**Pull request closed without merging**